APITestingPOC/TestSuite.cs

```diff
@@ -1,143 +1,94 @@
-            body = "Updated Body",
-            userId = 1
-        };
+            //Rest Client
 
-        request.AddJsonBody(requestBody);
-        var response = client.Execute(request);
-        var jsonResponse = JsonConvert.DeserializeObject<PostUserResponse>(response.Content);
+            var response = await apiHelper.PostAsync<User_PostApiResponse>("posts");
+            var jsonResponse = JsonConvert.DeserializeObject<User_PostApiResponse>(response.Content);
 
 
-        Assert.Equal(HttpStatusCode.Created, response.StatusCode);
-        Assert.NotNull(jsonResponse);
-        Assert.Equal("Updated Title", jsonResponse.title);
-        Assert.Equal("Updated Body", jsonResponse.body);
-        Assert.Equal(1, jsonResponse.userId);
-
-        int createdId = jsonResponse.id;
-        output.WriteLine("Request was successful. createdId: {0} " + createdId);
-    }
-    [Fact]
-    public void TestPutMethod()
-    {
-        var client = new RestClient("https://jsonplaceholder.typicode.com");
-        var request = new RestRequest("todos/{id}", Method.Put);
-        request.AddUrlSegment("id", 101);
+            Assert.Equal(HttpStatusCode.Created, response.StatusCode);
+            Assert.NotNull(jsonResponse);
+            Assert.Equal("Updated Title", jsonResponse.title);
+            Assert.Equal("Updated Body", jsonResponse.body);
+            Assert.Equal(1, jsonResponse.userId);
 
+            int createdId = jsonResponse.id;
+            output.WriteLine("Request was successful. createdId: {0} " + createdId);
+        }
 
-        var requestBody = new PutUserResponse
+        [Fact]
+        public async Task PutUserApiTest()
         {
-            id = 101,
-            title = "Updated Title",
-            completed = true
-        };
-
-        request.AddJsonBody(requestBody);
-        var response = client.Execute(request);
-        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            //Rest Client
 
-        var jsonResponse = JsonConvert.DeserializeObject<PutUserResponse>(response.Content);
+            var response = await apiHelper.PostAsync<User_PutApiResponse>("posts");
+            Assert.Equal(HttpStatusCode.Created, response.StatusCode);
 
-        Assert.NotNull(jsonResponse);
-        Assert.Equal(requestBody.id, jsonResponse.id);
-        Assert.Equal(requestBody.title, jsonResponse.title);
-        Assert.True(jsonResponse.completed);
+            var jsonResponse = JsonConvert.DeserializeObject<User_PutApiResponse>(response.Content);
 
-    }
-    [Fact]
-    public void TestDeleteMethod()
-    {
-        var client = new RestClient("https://jsonplaceholder.typicode.com");
-        var request = new RestRequest("todos/{id}", Method.Delete);
-        request.AddUrlSegment("id", 13);
+            Assert.NotNull(jsonResponse);
+            Assert.Equal(101, jsonResponse.id);
+            Assert.Equal("Updated Title", jsonResponse.title);
+            Assert.True(true);
```

Comment:
This line can be removed

Reply:
This asserts the response coming from the PUT API response, but instead of directly putting value, I'll try to assert the response.

Reply:
I have updated the line of code and pushed it. Please review the latest commit.